src/Core/Compiler/CompilerService.cs

```diff
@@ -283,20 +291,25 @@ public IEnumerable<QsNamespaceElement> IdentifyElements(string source, ITaskRepo
                 IsExecutable = compileAsExecutable,
                 AssemblyConstants = new Dictionary<string, string> { [AssemblyConstants.ProcessorArchitecture] = executionTarget ?? string.Empty },
                 TargetCapability = capability ?? TargetCapabilityModule.FullComputation,
+                // ToDo: The logic for determining if we should generate QIR should live in the SubmitterFactory as a
+                // static function, `TargetSupportsQir`. We should then be calling that static function here for
+                // PrepareQirGeneration instead of this logic.
+                // GitHub Issue: https://github.com/microsoft/qsharp-runtime/issues/968
```

Comment:
IIUC, this is where we will need to adapt when more targets are enabled for QIR generation support? Given that, we may at least want to move this into `AzureExecutionTarget` as a property there so that it's easier to make changes as needed, now that we're looking at adding more targets pretty soon.

Reply:
Discussed offline. This is a change we will make in the future.